scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+# + middleName - Middle name(s) of the User 
+# + honorificPrefix - Title 
+# + honorificSuffix - Suffix
+public type Name record {
+    string formatted?;
+    string givenName?;
+    string familyName?;
+    string middleName?;
+    string honorificPrefix?;
+    string honorificSuffix?;
+};
+
+# Represents the Custom Schema 
+#
+# + idpType - Identity provider type  
+# + isReadOnlyUser - Whether the user is read only
+# + userSource - Source of the user 
+# + photoUrl - URI of the user's photo  
+# + userAccountType - Type of the user account  
+# + askPassword - Whether the user is asked to provide a password
+public type Custom record {
+    string idpType?;
+    boolean|string isReadOnlyUser?;
+    string userSource?;
+    string photoUrl?;
+    string userAccountType?;
+    boolean askPassword?;
+};
+
+# Represents the request of updateUser operation.
+#
+# + schemas - URIs of the used SCIM schemas 
+# + externalId - Identifier by the provisioning client  
+# + userName - Unique identifier within the system
+# + name - Components of the user's name   
+# + displayName - Name displayed to end-users
+# + nickName - Casual name of the user  
+# + profileUrl - URI of the user's online profile   
+# + emails - Email addresses for the User 
+# + addresses - Physical mailing address for the user
+# + phoneNumbers - Phone numbers for the user  
+# + ims - Instant messaging addresses for the user  
+# + roles - List of roles for the user  
+# + photos - URI of image of the user  
+# + userType - Relationship between the organization and the user   
+# + title - Title of the user
+# + preferredLanguage - Preferred written or spoken languages of the user
+# + locale - Default location of the user  
+# + timezone - Time zone of the user  
+# + active - Administrative status of the user  
+# + groups - Groups to which the user belongs  
+# + x509Certificates - Certificates associated with the user  
+# + meta - Metadata of the user  
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema
+public type UserUpdate record {
+    string[]|string schemas?;
+    int externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    Email[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Ims[] ims?;
+    Role[] roles?;
+    Photo[] photos?;
+    string userType?;
+    string title?;
+    string preferredLanguage?;
+    string locale?;
+    string timezone?;
+    boolean active?;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+};
+
+# Represents the sub-attributes of the operations in userPatch record.
+#
+# + op - Operation to be performed  
+# + value - Values to be used for the operation
+public type PatchOperations record {
+    string op;
```

Comment:
Shall we define the allowed operations using an enum or any other suitable data type